File: bundles/org.eclipse.vjet.vjo.vjojavalib/bin/org/eclipse/vjet/vjo/java/util/Map.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
vjo.itype('org.eclipse.vjet.vjo.java.util.Map<K,V>') //< public
.needs('org.eclipse.vjet.vjo.java.util.Set','')
.needs('org.eclipse.vjet.vjo.java.util.Collection','')
.props({
    Entry:vjo.itype() //< public Entry<K,V>
    .protos({
        //> public K getKey()
        getKey:function(){
        },
        //> public V getValue()
        getValue:function(){
        },
        //> public V setValue(V value)
        setValue:function(value){
        },
        //> public boolean equals(Object o)
        equals:function(o){
        },
        //> public int hashCode()
        hashCode:function(){
        }
    })
    .endType()
})
.protos({
    //> public int size()
    size:function(){
    },
    //> public boolean isEmpty()
    isEmpty:function(){
    },
    //> public boolean containsKey(Object key)
    containsKey:function(key){
    },
    //> public boolean containsValue(Object value)
    containsValue:function(value){
    },
    //> public V get(Object key)
    get:function(key){
    },
    //> public V put(K key,V value)
    put:function(key,value){
    },
    //> public V remove(Object key)
    remove:function(key){
    },
    //> public void putAll(Map<? extends K,? extends V> t)
    putAll:function(t){
    },
    //> public void clear()
    clear:function(){
    },
    //> public Set<K> keySet()
    keySet:function(){
    },
    //> public Collection<V> values()
    values:function(){
    },
    //> public Set<Map<K,V>> entrySet()
    entrySet:function(){
    },
    //> public boolean equals(Object o)
    equals:function(o){
    },
    //> public int hashCode()
    hashCode:function(){
    }
})
.endType();

File: bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vjo/util/Map.java
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
package org.eclipse.vjet.vjo.util;

import java.util.HashMap;
import org.eclipse.vjet.dsf.javatojs.anno.AJsProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsProxy;
import org.mozilla.mod.javascript.Scriptable;
import org.eclipse.vjet.dsf.javatojs.anno.AExclude;
import org.eclipse.vjet.dsf.dap.proxy.INativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsTypeRef;
import org.eclipse.vjet.dsf.javatojs.anno.AJavaOnly;

//NativeJsProxy for vjo.util.Map.js
@org.eclipse.vjet.dsf.resource.utils.CodeGen("NativeJsProxyGenerator")
@AJsProxy
public class Map extends NativeJsProxy {

    /** for framework use only */
    @AExclude
    public Map(Scriptable nativeObj){
        super(nativeObj);
    }

    /** internal use only */
    protected Map(Object ...args){
        super(args);
    }

    public Map() {
        super();
    }

    public Map(Object psJavaClass) {
        super(psJavaClass);
    }

    public Map(HashMap psJavaClass) {
        super(psJavaClass);
    }

    public Object get(Object key) {
        return callWithName("get", Object.class, key);
    }

    public Object put(Object key, Object value) {
        return callWithName("put", Object.class, key, value);
    }

    public boolean remove(Object key) {
        return callWithName("remove", Boolean.class, key);
    }

    public int size() {
        return callWithName("size", Integer.class);
    }

    @AJavaOnly
    public static final NativeJsTypeRef<Map> prototype = NativeJsTypeRef.get(Map.class);

    public final INativeJsFuncProxy<Map> get = NativeJsFuncProxy.create(this, "get");

    public final INativeJsFuncProxy<Map> put = NativeJsFuncProxy.create(this, "put");

    public final INativeJsFuncProxy<Map> remove = NativeJsFuncProxy.create(this, "remove");

    public final INativeJsFuncProxy<Map> size = NativeJsFuncProxy.create(this, "size");
}